packages/datasource-customizer/src/decorators/actions/result-builder.ts

```
@@ -90,4 +90,13 @@ export default class ResultBuilder {
   redirectTo(path: string): ActionResult {
     return { type: 'Redirect', path };
   }
+
+  /**
+   * Request an approval from the code
+   * @example
+   * .requestApproval();
+   */
+  requestApproval(): ActionResult {
+    return { type: 'RequestApproval' };
+  }
```

Ok. We need a little more logic in there. I didn't want to throw the error in there.

But we will be required to do so. Because in the current state, the approval cannot be approved since it retriggers an approval. 🎅 

In this case `requestApproval` should be smart enough to only throw (request an approval) when it's not an approval.